src/RobotInterface/src/Helper.cpp

```diff
@@ -489,6 +504,21 @@ bool RobotInterface::setPositionReferences(const iDynTree::VectorDynSize& desire
             return false;
         }
         m_controlMode = VOCAB_CM_POSITION;
+
+        std::vector<yarp::dev::InteractionModeEnum>  modes;
+        std::vector<int>  jointsIndex;
+        for (int i=0;i<m_actuatedDOFs;i++) {
+            jointsIndex.at(i)=i;
+            if(m_isJointModeStiffVector.at(i)==true){
+            modes.at(i)=yarp::dev::InteractionModeEnum::VOCAB_IM_STIFF;
+            }
+            else {
+            modes.at(i)=yarp::dev::InteractionModeEnum::VOCAB_IM_COMPLIANT;
+            }
+        }
+
+        m_InteractionInterface->setInteractionModes(m_actuatedDOFs,jointsIndex.data(),modes.data());
```

Comment:
You can call [this](http://www.yarp.it/classyarp_1_1dev_1_1IInteractionMode.html#a856cb0d51a4c7c6e810ade9a5e0f77f6) function to change the interaction mode
`m_InteractionInterface->setInteractionModes(modes.data())`